src/nvim/os/win_defs.h

```diff
@@ -1,6 +1,8 @@
 #ifndef NVIM_OS_WIN_DEFS_H
 #define NVIM_OS_WIN_DEFS_H
 
+#ifdef WIN32
```

Comment:
this shouldn't be necessary. Could add a `STATIC_ASSERT(!defined(WIN32))` instead.